docs/source/multi_gpu.rst

```diff
@@ -50,7 +50,20 @@ This will make your code scale to any arbitrary number of GPUs or TPUs with Ligh
         z = torch.Tensor(2, 3)
         z = z.type_as(x, device=self.device)
 
-The LightningModule knows what device it is on. You can access the reference via `self.device`.
+The :class:`~pytorch_lightning.core.lightning.LightningModule` knows what device it is on. You can access the reference via `self.device`.
+Sometimes it is necessary to store tensors as module attributes. However, if they are not parameters they will
+remain on the CPU even if the module gets moved to a new device. To prevent that and remain device agnostic,
+register the tensor as a buffer in your modules's `__init__` method with :meth:`~torch.nn.Module.register_buffer`.
+
+.. testcode::
+
+    class LitModel(LightningModule):
+
+        def __init__(self):
+            ...
+            self.register_buffer("sigma", torch.eye(3))
+            # you can now access self.sigma anywhere in your module
```

Comment:
When we're already on it, we should maybe also add something to register parameters here?

Reply:
maybe we cand document the three types like this:

```python
# will stay on cpu unless moved manually
self.tensor = torch.eye(3) 

# always on same device as module
self.register_buffer("tensor", torch.eye(3))

# always on same device as module and will show up in self.parameters()
self.tensor = torch.nn.Parameter(torch.eye(3))
```
What do you think? I was hesitant to document this in detail since it is not specific to Lightning, just pure PyTorch.